### ISSUE TITLE
When we navigate to "Auto Size" dropdown using the up/down arrow keys, it is getting auto selected without hitting ENTER key:A11y_ A11y_.NET CoreWinforms_On Focus.

### ISSUE THREAD

**Issue by AnKushSingh05** [dotnet/winforms#12031]
### .NET version

dotnet-sdk-9.0.100-preview.5.24258.1

### Did it work in .NET Framework?

.NET Framework and the InProc designer work the same as .NET+OOP designer 

### GitHub Tags:
#A11yMAS; #A11yTCS; #A11ySev2; #BM_.NET Core[Winforms]_Win32_May2024; #.NETCore; #WCAG3.2.1; #DesktopApp; #Win11;#keyboard;#On Focus;

### Environment Details:
App name: .NET Core Winforms
Window Version: Win 11
OS Build: 22621.3085
Screen readers: Narrator, NVDA 2023.3.3

### Repro Steps:

1. Launch the "Accessibility core app".
2. Main Form screen would appear.
3. TAB to "Remaining_Controls_Testing the controls under Printing Tab" and hit ENTER key.
4. "Remaining_Controls" popup will appear.
5. Tab till Auto size dropdown under layout section.
6. Observe that when we navigate to "File type" dropdown using the up/down arrow keys, it is getting auto selected without hitting ENTER key

### Actual Result:
When we navigate to "Auto Size" dropdown using the up/down arrow keys, it is getting auto selected without hitting ENTER key
### Expected Results:
When we navigate to "Auto Size" dropdown using the up/down arrow keys it should be selected after hitting ENTER key.
### User Impact:
User who relies on keyboard will face difficulty as they might select the control which they did not intent.

### Attachment

https://github.com/user-attachments/assets/5a649900-2ce7-4aa0-9e48-4b8e94ec18b4

**Comment by AnKushSingh05** [comment]
Primary bug: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2164329

**Comment by Zheng-Li01** [comment]
There have the same behaviors between .NET Framework and .NET on both Designer and Run time as below screenshot.
The user can directly select the value of **Auto Size** by using up/down arrow keys dropdown panel, the issue cannot reproduce other properties.
**.NET Framework:**
![FrameworkResults](https://github.com/user-attachments/assets/12aa564e-45df-4922-bf71-2bb40c23531d)

**.NET:**
![CoreResults](https://github.com/user-attachments/assets/b6cebc6a-d4b1-46d6-9b47-1c86e435d3ad)

**Comment by merriemcgaw** [comment]
We will need to fix this in .NET 10 since we don't have time to make it in .NET 9. We should work with the Accessibility team to get this added to an accessibility plan allowing us to fix it in .NET 10. I don't think there's enough benefit to warrant servicing .NET 9 unless we get customer requests.

Need to ask about GitHub tags in Office Hours today.

**Comment by merriemcgaw** [comment]
@Zheng-Li01 can you confirm this is the same behavior across all of the "combo boxes" in the PropertyGrid?

I will be filing AccessibilityPlan for the .NET WinForms PropertyGrid and a second one for VS. The .NET one will need to be addressed in .NET 10 and cover at least the ComboBoxe types of TypeEditors that we own. The VS plan will be separate.

**Comment by Zheng-Li01** [comment]
@merriemcgaw, go throughed all of the "combo boxes" in the PropertyGrid for Form, Lable, Button, ToolStrip, GroupBox etc. controls. the issue only can reproduce on **Auto Size** property, not reproduce on other properties.

**Comment by merriemcgaw** [comment]
@Zheng-Li01 can you see if it repros on a custom PropertyGrid with a bool type of property in it?

**Comment by Zheng-Li01** [comment]
> [@Zheng-Li01](https://github.com/Zheng-Li01) can you see if it repros on a custom PropertyGrid with a bool type of property in it?

Not reproduce on a custom PropertyGrid with a bool type of property in it as below screenshot.
![Image](https://github.com/user-attachments/assets/fbf96c73-6597-453e-9679-e63a8907e7bc)

**Comment by merriemcgaw** [comment]
@LeafShi1 can you please go ahead and fix this issue? We will create another issue to track improving the keyboard usability of the PropertyGrid as a feature for .NET 10.

**Comment by LeafShi1** [comment]
The following properties also have this issue:

- [ ] The `Multiline ` property of the TextBox
- [ ] The `Alignment ` property of the TabControl
- [ ] The `Checked` and `CheckState` property of the CheckBox
- [ ] The `RowHeadersWidthSizeMode` and `ColumnHeadersHeightSizeMod`e of the DataGridView
- [ ] The `Frozen` of the DataGridViewColumn

**Comment by Tanya-Solyanik** [comment]
@Zheng-Li01 @LeafShi1 I'm not sure how the to navigate property grid in this scenario. Here is what i'm doing
0. focus is on a label on a form in the designer
1. ALt+ENter
2. user up and down arrows to navigate to autosize property name
3. tab to the edit field on the right hand side
4. press an up or  down arrow to change the value.
result: value is changed, size property is updated in  the property grid and on Ctrl+S .designer.cs file is updated correctly. 
This seems a convenient flow to me. But we want to fix something else?

**Comment by LeafShi1** [comment]
> [@Zheng-Li01](https://github.com/Zheng-Li01) [@LeafShi1](https://github.com/LeafShi1) [@merriemcgaw](https://github.com/merriemcgaw) I'm not sure if the way described in the bug is the est way to navigate type editor with the keyboard. Here is what i'm doing 0. focus is on a label on a form in the designer
> 

After the change, we can still press the up and down keys on the edit box to change the value.
It's just that before the change, the edit box loses focus after pressing the up and down keys, but it will not after the change.

The whole operation process is some linkage behavior, and it does not cause context change.

Before Changes:
![Image](https://github.com/user-attachments/assets/db9d33a6-45df-4997-8817-00497e006996)

After Changes:
![Image](https://github.com/user-attachments/assets/01bc4531-0116-4fbf-bb82-d9dc060df42a)

**Comment by Tanya-Solyanik** [comment]
@merriemcgaw  - do you agree with #A11ySev2; assessment? Keyboard user does not have to open the drop down at all to perform these changes.

**Comment by merriemcgaw** [comment]
@Tanya-Solyanik yeah, I do. This is an example of the type of thing the SMEs were worried about us needing to fix. The value shouldn't be committed until the user takes action to hit enter or leave the field, IMO.

**Comment by AnKushSingh05** [comment]
@LeafShi1 Could you please provide the environment to verify the issue?. Thanks!

**Comment by Syareel-Sukeri** [comment]
The issue has been verified in the latest .NET 10 SDK build: .NET 10.0.100-alpha.1.24556.12 and is now resolved. The value is no longer being auto selected, and focus is maintained when using the up/down arrow keys, both when opening the dropdown or without doing so.

https://github.com/user-attachments/assets/a3753cd2-4a9c-4f99-91a1-23a3edffbb7a

**Comment by AnKushSingh05** [comment]
> The issue has been verified in the latest .NET 10 SDK build: .NET 10.0.100-alpha.1.24556.12 and is now resolved. The value is no longer being auto selected, and focus is maintained when using the up/down arrow keys, both when opening the dropdown or without doing so.
> 
>  DropDownFix.mp4

@Syareel-Sukeri Can you please give the environment link

**Comment by Syareel-Sukeri** [comment]
@AnKushSingh05 You can access the Accessibility-Core-App and .NET 10 for testing from this link: https://microsoft-my.sharepoint.com/:f:/p/v-smdsukeri/EmvP63iKbZBOq0Mwe4LY8uEBBfIjniIzd3ma660aVSp17w?e=Ahd0vc

**Comment by AnKushSingh05** [comment]
https://github.com/user-attachments/assets/080c9f51-1c04-4901-a2f4-cc1347a6df6e
Verified the issue in  .NET 10 SDK build: .NET 10.0.100-alpha.1.24556.12
Issue is fixed. Hence closing the bug
